doppler/api.go

```diff
@@ -59,7 +59,7 @@ func isSuccess(statusCode int) bool {
 	return (statusCode >= 200 && statusCode <= 299) || (statusCode >= 300 && statusCode <= 399)
 }
 
-func (client APIClient) GetRequest(ctx context.Context, path string, params []QueryParam) (*APIResponse, *APIError) {
+func (client APIClient) GetRequest(ctx context.Context, path string, params []QueryParam) (*APIResponse, error) {
```

Comment:
Good question -- No, we still want to return our custom `APIError` struct (so that it can contain the appropriate message and context). However, it's recommended that we keep that implementation detail localized to our function.

> It's a good idea for functions that return errors always to use the error type in their signature (as we did above) rather than a concrete type such as `*MyError`, to help guarantee the error is created correctly. As an example, [os.Open](https://go.dev/pkg/os/#Open) returns an error even though, if not nil, it's always of concrete type [*os.PathError](https://go.dev/pkg/os/#PathError).

https://go.dev/doc/faq#nil_error